test/export/StructureDefinitionExporter.test.ts

```diff
@@ -2294,6 +2410,620 @@ describe('StructureDefinitionExporter', () => {
+      rootFlag.mustSupport = false;
+
+      observationWithSlice.rules.push(labFlag, rootFlag);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootComponent = sd.findElement('Observation.component');
+      const labComponent = sd.findElement('Observation.component:Lab');
+      expect(rootComponent.mustSupport).toBe(false);
+      expect(labComponent.mustSupport).toBe(true);
+    });
+
+    it('should apply a FlagRule on the child of a sliced element that updates the flags on the child of a slice', () => {
+      // * component[Lab].interpretation 0..1 // this forces the creation of the unfolded slice
+      // * component.interpretation MS
+      const labCard = new CardRule('component[Lab].interpretation');
+      labCard.min = 0;
+      labCard.max = '1';
+      const rootFlag = new FlagRule('component.interpretation');
+      rootFlag.mustSupport = true;
+
+      observationWithSlice.rules.push(labCard, rootFlag);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootInterpretation = sd.findElement('Observation.component.interpretation');
+      const labInterpretation = sd.findElement('Observation.component:Lab.interpretation');
+      expect(rootInterpretation.mustSupport).toBe(true);
+      expect(labInterpretation.mustSupport).toBe(true);
+    });
+
+    it('should not apply a flag rule that would disable a must support or is modifier flag on a slice when the list has that flag', () => {
+      // * component.interpretation MS
+      // * component[Lab].interpretation ~MS
+      const rootFlag = new FlagRule('component.interpretation');
+      rootFlag.mustSupport = true;
+      const labFlag = new FlagRule('component[Lab].interpretation');
+      labFlag.mustSupport = false;
+      observationWithSlice.rules.push(rootFlag, labFlag);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootInterpretation = sd.findElement('Observation.component.interpretation');
+      const labInterpretation = sd.findElement('Observation.component:Lab.interpretation');
+      expect(rootInterpretation.mustSupport).toBe(true);
+      expect(labInterpretation.mustSupport).toBe(true);
+    });
+
+    // don't actually apply the rule to the slice, since it might be valid based on the contents of the value sets
+    // but if it's the same value set for both, _then_ we know it can't be a weaker binding, and that's what we should give a message about
+    // so this is ok:
+    // * category[Procedure] from ABC (extensible)
+    // * category from ABC (required)
+    // and it will update the strength to required
+    // this is not:
+    // * category from ABC (required)
+    // * category[Procedure] from ABC (extensible)
```

Comment:
Is this comment out of place?

Reply:
This just looks like a note I left for myself and forgot to remove. oops